spec/unit/provider/oneview_server_hardware_c7000_spec.rb

```diff
@@ -0,0 +1,204 @@
+################################################################################
+# (C) Copyright 2016-2017 Hewlett Packard Enterprise Development LP
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# You may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+################################################################################
+
+require 'spec_helper'
+
+provider_class = Puppet::Type.type(:oneview_server_hardware).provider(:c7000)
+api_version = login[:api_version] || 200
+resource_name = 'ServerHardware'
+resourcetype = if api_version == 200
```

Comment:
Please help me understand. Should we add this verification only for the resources which have differences across API versions?